evidence/pool.go

```diff
@@ -0,0 +1,80 @@
+package evidence
+
+import (
+	"github.com/tendermint/tmlibs/log"
+
+	cfg "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/types"
+)
+
+// EvidencePool maintains a pool of valid evidence
+// in an EvidenceStore.
+type EvidencePool struct {
+	config *cfg.EvidenceConfig
+	logger log.Logger
+
+	state         types.State
+	evidenceStore *EvidenceStore
+
+	evidenceChan chan types.Evidence
+}
+
+func NewEvidencePool(config *cfg.EvidenceConfig, evidenceStore *EvidenceStore, state types.State) *EvidencePool {
+	evpool := &EvidencePool{
+		config:        config,
+		logger:        log.NewNopLogger(),
+		evidenceStore: evidenceStore,
+		state:         state,
+		evidenceChan:  make(chan types.Evidence),
+	}
+	return evpool
+}
+
+// SetLogger sets the Logger.
+func (evpool *EvidencePool) SetLogger(l log.Logger) {
+	evpool.logger = l
+}
+
+// EvidenceChan returns an unbuffered channel on which new evidence can be received.
+func (evpool *EvidencePool) EvidenceChan() chan types.Evidence {
```

Comment:
How about encoding the direction of the channel here. 
`func (evpool *EvidencePool) EvidenceChan() <-chan types.Evidence`

https://gobyexample.com/channel-directions